src/Mvc/src/Microsoft.AspNetCore.Mvc.RazorPages/CompiledPageActionDescriptor.cs

```diff
@@ -59,5 +60,10 @@ public CompiledPageActionDescriptor(PageActionDescriptor actionDescriptor)
         /// Gets or sets the <see cref="TypeInfo"/> of the page.
         /// </summary>
         public TypeInfo PageTypeInfo { get; set; }
+
+        /// <summary>
+        /// Gets or sets the associated <see cref="Endpoint"/> of this page.
+        /// </summary>
+        public Endpoint Endpoint { get; set; }
```

Comment:
The compiled page AD now always has an endpoint as well. This isn't super fun but it seems like the most reasonable thing to do.